internal/controller/lustrefilesystem_controller.go

```diff
@@ -86,6 +86,20 @@ func (r *LustreFileSystemReconciler) Reconcile(ctx context.Context, req ctrl.Req
 			return ctrl.Result{}, nil
 		}
 
+		// Check to see that only the lustre filesystem finalizer exists
+		onlyLustreFinalizer := func(o client.Object) bool {
+			f := o.GetFinalizers()
+			if len(f) == 1 && f[0] == finalizerLustreFileSystem {
```

Comment:
We got past line 85, so we've already confirmed that the finalizer owned by this controller is present.  I think, before line 85 you want to know if there are other finalizers, and return if they are present.